### PR TITLE
[#56]: Prepare backend

### PR DESCRIPTION
**Pull request by worm2fed**
## Description

Problem:
Backend app was very empty, with default definitions from template. We
need to prepare it for development process.
    
Solution:
This PR is about defining initial architecture, so as result we
have some monads to work with.
For now we have application monad where we can read config at startup
and store it in env along with DB pool connection.
Also we have simple API to return current config, it will be deleted and
now exists only to show that we can server API.
We have some initial error types and some core definitions to use
monitoring and jwt (jwt tests presented).

## Related issue(s)

#56 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

- Public contracts
  - [x] My changes doesn't break any Licenses.
  - [x] I added an entry to the [changelog](../tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Union](https://github.com/union-platform/guides/blob/master/commits.md).
- [x] My code complies with the [styles guide](https://github.com/union-platform/guides/blob/master/README.md#programming-languages).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 [![License: AGPL-3.0](https://img.shields.io/badge/License-AGPL%203.0-brightgreen.svg)](https://opensource.org/licenses/AGPL-3.0)
 
-Union is a platform where you can find your favorite matier, join a team, create something important or crazy; this is a place where you can be yourself.
+Union is a platform where you can find your favorite matier, join a team, create
```
Comment (rexolion) [inline README.md:11]: What is «matier» :) Maybe "find a mate" or "find a co-founder" or smth like that?

**Comment by rexolion** [comment]
Nice, where we will store docs for backend API?

**Comment by worm2fed** [comment]
@rexolion we'll have swagger, in this PR you can find very simple scratch for it in Core.

**Review by rexolion — APPROVED** [review]
Nice job

File: README.md (hunk 1 above)
Comment (worm2fed) [reply README.md:11]: didn't noticed your comment
let's change it in future work

**Pull request merged**